recipe/meta.yaml

```diff
@@ -32,47 +27,49 @@ requirements:
     # directly pin boost-cpp as we also seem to directly include boost symbols
     # in the Python modules.
     - arrow-cpp {{ version }}
-    - boost-cpp
-    - cython
-    - grpc-cpp
+    - boost-cpp 1.73.0
+    - cython 0.29.32
+    - grpc-cpp 1.46.1
     - numpy
     - pip
     - python
     - setuptools
     - setuptools_scm
-    - uriparser
+    - uriparser 0.9.3
     - wheel
   run:
-    - {{ pin_compatible('arrow-cpp') }}
+    # - {{ pin_compatible('arrow-cpp') }}
+    - arrow-cpp >=10.0.0,<11.0a0
```

Comment:
I think this can be done with pin_compatible(... min_pin='x') or something like that? [Docs](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html?highlight=pin_compatible#extra-jinja2-functions)

Reply:
I ended up removing this and instead rely on arrow-cpp `run_exports`